Minio/HttpRequestMessageBuilder.cs

```diff
@@ -78,6 +78,7 @@ public HttpRequestMessage Request
                         case "content-type":
                             try
                             {
+                                val ??= "application/octet-stream";
```

Comment:
We can also discuss this in the meeting with @abperiasamy, @kannappanr and @harshavardhana.

Reply:
When I look at this code, customers can set the `Content-Type` header, but if they set it to an empty value then it will be replaced with `application/octet-stream`. If you set the `Content-Type` header, then the user should set it to a valid value or the call is allowed to fail. If you don't want to set it, then just leave out the value.

This PR contains "fixes" that are well beyond the scope of the PR. So I would rollback this change, unless there is a good reason to do this.